handler/src/test/java/io/netty/handler/ssl/SSLEngineTest.java

```diff
@@ -236,15 +276,15 @@ private void mySetupMutualAuth(
             File clientTrustCrtFile, File clientKeyFile, File clientCrtFile, String clientKeyPassword)
             throws InterruptedException, SSLException {
         serverSslCtx = SslContextBuilder.forServer(serverCrtFile, serverKeyFile, serverKeyPassword)
-                .sslProvider(sslProvider())
+                .sslProvider(sslServerProvider())
                 .trustManager(servertTrustCrtFile)
                 .ciphers(null, IdentityCipherSuiteFilter.INSTANCE)
                 .sessionCacheSize(0)
                 .sessionTimeout(0)
                 .build();
 
         clientSslCtx = SslContextBuilder.forClient()
-                .sslProvider(sslProvider())
+                .sslProvider(sslClientProvider())
```

Comment:
`cleanupSslContext` assumed both context are generated using the same sslProvider. Consider breaking this into 2 methods as well.